ambry-api/src/main/java/com/github/ambry/config/Http2ClientConfig.java

```diff
@@ -167,6 +169,22 @@ public class Http2ClientConfig {
   @Default("false")
   public final boolean http2TimeoutAsNetworkError;
 
+  /**
+   * Denotes how much extra time should be given for requests to receive response under heavy load.
+   */
+  @Config(HTTP2_REQUEST_ADDITIONAL_TIMEOUT_MS)
+  @Default("2500")
+  public final int http2RequestAdditionalTimeoutMs;
+
+  /**
+   * Denotes the number of requests to consider when increasing the network time out. For example, if
+   * {@link this#http2RequestAdditionalTimeoutMs} is 2500 and this value is 100000, it means that we will
+   * increment network timeout by 2.5 seconds for every 100K requests.
+   */
+  @Config(HTTP2_REQUEST_COUNT_FOR_SCALING_TIMEOUT)
+  @Default("100000")
+  public final int http2RequestCountForScalingTimeout;
```

Comment:
Can we use chunkCount instead of request count? Also make it explicit that this is for update ttl only.

Reply:
Hmm, similar as above comment. Since we are using this configuration at frontend/server network layer which only knows of request-responses, wondering if `request count` is still okay.